### PR TITLE
Relevance checker

### PR DESCRIPTION
**Pull request by jamesgorrie**
~Fork of #4374~

Completely not prod ready, but behind a toggle to see how we might start integrating with logging stack.

Sends off an event which say which service to integrate with (the indexes for `search_logs` and `relevance_rating` are different as the querying etc is very different). Sends of ID / position / rating.

Seems like a relatively small amount of work.

It'd be good to get this is as it's needed to test end to end.

Screenshot 2019-04-11 at 17 42 49

### CODE REVIEW

File: common/views/components/RelevanceRater/RelevanceRater.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    service: 'relevance_rating',
+    event: 'Rate Result Relevance',
+    rating: rating,
+  };
+}
+
+const RelevanceRater = ({ id, position }: Props) => {
+  return (
+    <RelevanceRaterStyle>
+      <RelevanceRating
```
Comment (davidpmccormick) [inline common/views/components/RelevanceRater/RelevanceRater.js]: Could we find a simple way to indicate whether `1` or `4` is most relevant?

Comment (jamesgorrie) [reply common/views/components/RelevanceRater/RelevanceRater.js]: The interface / wording etc really needs work. @hthair  does have a key with with words that make more sense.

This is just to see if it works.

FYI 1 is irrelevant / 4 relevant.

Comment (jamesgorrie) [reply common/views/components/RelevanceRater/RelevanceRater.js]: We'll be making a ticket for this and we can capture all the piece that need doing.

**Comment by jamesgorrie** [comment]
@davidpmccormick probably easier to review this work after #4374 has gone in.

**Comment by jamesgorrie** [comment]
Ping @harrisonpim

**Comment by jamesgorrie** [comment]
Potentially related to https://github.com/wellcometrust/platform/issues/3569

**Comment by jamesgorrie** [comment]
This has now changed the tracking to also be less "search" specific.

The shape of the events is:
```
{
  service: string,
  name: string,
  data: Object
}
```

Currently we have two services with the following events:
* `search`
  * `Search` : someone made a search. Has properties:  `query`, `page`, `workType`, `_queryType`
  * `Select Result` : someone chose a result. Has properties: `query`, `page`, `workType`, `_queryType`, `position`
* `relevance_rating`
  * `Rate Result Relevance` : someone gave us a rating on relevance. Has Properties: `query`, `page`, `workType`, `_queryType`, `position`, `id`, `rating`

**Comment by jamesgorrie** [comment]
@wellcometrust/experience-devs the event structure will now be more like this:
https://github.com/wellcometrust/platform/issues/3569#issuecomment-484193702

**Comment by jamesgorrie** [comment]
Potential view of the es document:
```
{
  "_id": "ajs-f57a3d2d12ea5b4f314ebec9417bd24d",
  "anonymousId": "df458b4e-2e27-42a6-b163-fb443548a98a",
  "network": "StaffCorporateDevices",
  "toggles": {
    "toggle_amazingTings": true,
    "toggle_noggle": false
  },
  "data": {
    "event": "Result relevance rating",
    "rating": 3,
    "query": "horses",
    "page": 2,
    "workType": ["k", "q"],
    "_queryType": null,
  }
}
```

**Pull request merged**